### ISSUE TITLE
[BUG_SP] Worker drives off field when driving to first waypoint

### ISSUE THREAD

**Issue by Martin-L45** [Courseplay/Courseplay_FS22#3048]
**READ FIRST**
https://github.com/Courseplay/Courseplay_FS22/wiki

**Steps to reproduce**
Steps to reproduce the behavior.

**Actual behavior**
A clear and concise description of what happen.
In version 7.402 when driving to first waypoint the worker drives off the field. In version 7.4.0.1 this does not happen. The pictures are both versions started from the same savegame 

**Expected behavior**
Stay in the field
**Screenshots**

Version 7.4.0.1
![Farming Simulator 22 Screenshot 2024 01 03 - 17 34 05 80](https://github.com/Courseplay/Courseplay_FS22/assets/85753728/2d2a8287-a204-4c0d-8c03-e39ece0fbef1)

Version 7.4.0.2

![Farming Simulator 22 Screenshot 2024 01 03 - 17 36 24 94](https://github.com/Courseplay/Courseplay_FS22/assets/85753728/ad4cbb7c-d509-4a24-bb6a-fef30a5cf712)

![Farming Simulator 22 Screenshot 2024 01 03 - 17 37 18 38](https://github.com/Courseplay/Courseplay_FS22/assets/85753728/b6a98d52-704f-497c-82da-202da677dc22)

**Courseplay version**
Version used: 

**Log**
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/13821380/log.txt)

Please attach a log with all debug channels active.
For help with the debug channels,
please check out the readme at the github front page.

**Map**
Map played on.

**Vehicles/ implements involved**
Which vehicle or implement?
Mod yes/no ?

**Relevant script mods**
Mods like Autodrive or VCA and so on.

**Comment by Martin-L45** [comment]
I just did a test with this fix and now it works!!
Awesome work guys

**Comment by coatsy35** [comment]
Did the avoid fruit logic get altered when this was fixed? seeing issues where the trailer will not drive to the unloader when it returns to field if the route around is too far. See imagine below, the unloader was just sat at the entrance for 10 mins, when I turned on allow drive in fruit and stopped and started it, it found a route straight through the fruit and unloaded the combine as you would expect. I don't ever have drive on fruit turned on usually and I've not seen this before, it obviously fails to find a way around the fruit or times out perhaps? these are large fields. Maybe nothing but thought i'd mention it.

![fsScreen_2024_01_14_20_12_25](https://github.com/Courseplay/Courseplay_FS22/assets/122838698/895e6c44-0168-470f-a1a6-c815e3099886)
